tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala

```diff
@@ -0,0 +1,69 @@
+package mesosphere.marathon.integration
+
+import java.util.UUID
+
+import mesosphere.AkkaIntegrationTest
+import mesosphere.marathon.integration.facades.MarathonFacade.extractDeploymentIds
+import mesosphere.marathon.integration.setup.EmbeddedMarathonTest
+import mesosphere.marathon.raml.{App, AppCheck, CommandCheck, ShellCommand}
+import mesosphere.marathon.state.PathId
+
+class MesosCheckIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
+
+  def appId(suffix: Option[String] = None): PathId = testBasePath / s"app-${suffix.getOrElse(UUID.randomUUID)}"
+  val appCommand: String = s"sleep 100"
+
+  "AppDeploy with mesos checks" should {
+    "provide a command check result status of 0" in {
+      Given("a new app with checks")
+      val id = appId(Some("with-successful-checks"))
+      val app = App(
+        id.toString,
+        cmd = Some(appCommand),
+        check = Some(ramlCommandCheck("true")))
+
+      When("The app deploys")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created with a successful exit code")
+      result should be(Created)
+      extractDeploymentIds(result) should have size 1
+      waitForDeployment(result)
```

Comment:
Feels like we should have another `Then` here? 

` Then("The app is created")`

`And("the successful exit code is exposed via tasks")`